pkg/controllers/managedos/template.go

```diff
@@ -165,10 +166,12 @@ func (h *handler) objects(mos *elm.ManagedOSImage) ([]runtime.Object, error) {
 
 	upgradeContainerSpec.Env = metadataEnv
 
+	objectName := name.SafeConcatName("os-upgrader", mos.Name, string(mos.UID))
```

Comment:
Having this line (almost) duplicated from `managedos.go`, I'd rather see this refactored to something like
`BuildName("os-upgrader")` (and `BuildName("mos")`), where `BuildName` takes care of adding the right artefacts.

Reply:
nope. Because I need to pass the ManagedOSImage, so we end up with a function that needs testing and its composed of  10 lines for something that takes 2 lines.

DRY is good when it fits, not here :)